Update GitBook structures (#27)
yuhattor/japanese-contents: Getting-Started-with-InnerSource/README.md

# インナーソース入門

企業におけるコラボレーションと生産性を高める秘訣

![インナーソース入門][cover-image]

**原題:** [Getting Started with InnerSource](https://innersourcecommons.org/learn/books/getting-started-with-innersource/)<br />
**著者:** Andy Oram<br />
**発行日:** 2015年7月<br />
**発行元:** O’Reilly Media<br />
**翻訳:** InnerSource Commons Japan

## まえがき

インナーソースの世界へようこそ。インナーソースは、閉じた組織におけるオープンソースソフトウェアを開発するための原動力であり、プロジェクトの "オープン性" を企業内のチームを超えて広げるものです。この本では、オライリーの編集者兼著者である Andy Oram が、オープンソース開発を成功させる原則を確認するとともに、大手 EC の PayPal でインナーソースがどのように機能したのかを説明することによって、皆様をインナーソースの世界に導きます。

> PayPal におけるインナーソースのパスは、ツールや企業文化の意識的なシフトを含む大企業の一連の意思決定を含んでいました。インナーソースを通じて、同社はより迅速な開発とより良い品質を達成するだけでなく、プログラマが PayPal のどの開発プロジェクトにも貢献することを奨励し、チーム間の共創を可能にしました。

Reviewer: 本文によく似た文がありますので、そちらに合わせます。大きくは変わりません。

> PayPal が InnerSource を採用するまでの過程には企業としての意志決定の積み重ねがあり、歴史的で大規模なものです。
> InnerSource の採用は、それまでに何度かあった企業変革と同様に、ツールと企業文化の戦略的変更として進められました。

```suggestion
PayPal がインナーソースを採用するまでの過程には企業としての意思決定の積み重ねがあり、その中にはツールと文化の意識的な刷新が含まれています。
インナーソースを通じて、同社はより迅速な開発とより良い品質を達成するだけでなく、プログラマが PayPal のどの開発プロジェクトにも貢献することを奨励し、チーム間の共創を可能にしました。
```

この本を通して読者の皆様は、以下のようなインナーソース戦略の具体的なメリットを学ぶことができます。

* より高速な開発: プログラマーは、ユニットテスト、コードカバレッジ、および継続的インテグレーションを使用して、初期の段階でバグを取り除くことができます。
* 完全なドキュメント化: コード内のコメントやディスカッションリストでの形式的なものでなく、より良いドキュメントが作成されます。
* コードの再利用: 組織全体のプログラマーが、他のチームが開発したモジュールのコードとアーキテクチャを理解することができます。
* チーム横断的なコラボレーション: チーム外からの貢献には摩擦がなくなり、書き直されることがほとんどなくなります。
* GitHub を使った開発: GitHub では、社内プロジェクト用のプライベートリポジトリと、オープンソースコード用のパブリックリポジトリを管理します。

> この書籍は、クリエイティブ・コモンズ・ライセンス(CC BY)のライセンスにより提供されています。
> どなたでも無料でこの書籍を手に取れるほか、[原本](https://innersourcecommons.org/documents/books/GettingStartedWithInnerSource.pdf)をPDF形式（3.3 MB）でダウンロードすることができます。

Reviewer: GitBook の本に対して手に取れるとは表現しないと考え、以下の訳を考えました。

```suggestion
この書籍は無償であり、クリエイティブ・コモンズ・ライセンス(CC BY)で提供されています。
[原本 (PDF形式3.3 MB)](https://innersourcecommons.org/learn/books/getting-started-with-innersource/)も同じく CC BY ライセンスで提供されており、誰でもダウンロードすることができます。
```

# インナーソース入門

企業におけるコラボレーションと生産性を高める秘訣

![インナーソース入門][cover-image]

**原題:** [Getting Started with InnerSource](https://innersourcecommons.org/learn/books/getting-started-with-innersource/)<br />
**著者:** Andy Oram<br />
**発行日:** 2015年7月<br />
**発行元:** O’Reilly Media<br />
**翻訳:** InnerSource Commons Japan

## まえがき

インナーソースの世界へようこそ。インナーソースは、閉じた組織におけるオープンソースソフトウェアを開発するための原動力であり、プロジェクトの "オープン性" を企業内のチームを超えて広げるものです。この本では、オライリーの編集者兼著者である Andy Oram が、オープンソース開発を成功させる原則を確認するとともに、大手 EC の PayPal でインナーソースがどのように機能したのかを説明することによって、皆様をインナーソースの世界に導きます。

PayPal がインナーソースを採用するまでの過程には企業としての意思決定の積み重ねがあり、その中にはツールと文化の意識的な刷新が含まれています。
インナーソースを通じて、同社はより迅速な開発とより良い品質を達成するだけでなく、プログラマが PayPal のどの開発プロジェクトにも貢献することを奨励し、チーム間の共創を可能にしました。

この本を通して読者の皆様は、以下のようなインナーソース戦略の具体的なメリットを学ぶことができます。

* より高速な開発: プログラマーは、ユニットテスト、コードカバレッジ、および継続的インテグレーションを使用して、初期の段階でバグを取り除くことができます。

> * 完全なドキュメント化: コード内のコメントやディスカッションリストでの形式的なものでなく、より良いドキュメントが作成されます。

Reviewer: 本編に似た記述がありますので、これとできるだけ同じ意味になるように仕上げます。

> Programmers learn to document their code better, both formally (as in-code comments and documentation) and informally (on discussion lists).

また、ディスカッションリストだと伝わらないでしょうから、チャット・チケット・メーリングリストと明示します。

```suggestion
* 完全なドキュメント化: コードを説明するドキュメントの質が向上します。コードに併記される公式なコメントの質が上がることと、チャット・チケット・メーリングリストでの議論で公式的な説明を控えることの両方を含みます。
```

`less formally on discussion lists` に関する説明が本編からは見当たらず「控える」と表現しましたが、少なくとも矛盾はしないと思います。おそらく内容としては、公式的な説明はドキュメントに書くべきで、チャットやメーリングリストといった流れ去る場所には書かないようにするのがよいという意味だと思うのですが、そこが説明されていません。

* コードの再利用: 組織全体のプログラマーが、他のチームが開発したモジュールのコードとアーキテクチャを理解することができます。
* チーム横断的なコラボレーション: チーム外からの貢献には摩擦がなくなり、書き直されることがほとんどなくなります。

> * GitHub を使った開発: GitHub では、社内プロジェクト用のプライベートリポジトリと、オープンソースコード用のパブリックリポジトリを管理します。

Reviewer: ささいですが、private/public 両方のリポジトリを扱えることを書くとメリットらしくなります。

```suggestion
* GitHub を使った開発: GitHub では、社内用のプライベートリポジトリだけでなく、OSS 用のパブリックリポジトリも合わせて扱うことができます。
```

> # 本書について

Reviewer: 「本書について」 よりも 「インナーソース入門」 を大きく表示させたいところです。

README.md の構成を `# インナーソース入門` だけとして `## インナーソース入門` と `## あとがき` を消し、

SUMMARY.md をこうすれば改善するでしょうか。

```
* [インナーソース入門](./README.md)
* [本編](./ja.md)
```

Author: これは、なやみました。
SUMMARY.md は目次の役割も果たすほか、章のタイトルにもなってしまうんですよね。

そもそも、オライリーの区切り方は以下のようになっています。
h1 チーム間コラボレーションへの強力なアプローチ
┣ h2 オープンソースの原則が機能する場
┗ h2 PayPal が InnerSource を採用するまで

になっているんですよね。そもそも Heading が一個ずれています。
そしたらば、

```
* インナーソース入門 (Readme)
* h1 チーム間コラボレーションへの強力なアプローチ (ja)
* あとがき(afterword)
```
のほうがいいですかね

Author: ただ、これだと「チーム間コラボレーションへの強力なアプローチ」をみて本編だと気づかない可能性もあります。
グルーピングは可能なので、
本編
| h1 チーム間コラボレーションへの強力なアプローチ
|┣ h2 オープンソースの原則が機能する場
|┗ h2 PayPal が InnerSource を採用するまで

は可能かもしれません。ちょっと違うブランチで実験してみます

## インナーソース入門

企業におけるコラボレーションと生産性を高める秘訣

> ![インナーソース入門][cover-image]

Reviewer: 原著の表紙を流用できるのならこのままでよいと思います。

Author: 一旦これで Andy Oram 氏に連携をとってみようとおもいます！

> **原題:** Getting Started with InnerSource<br />

Reviewer: 原著へのリンクが欲しいところです。たとえばここに。

```suggestion
**原題:** [Getting Started with InnerSource](https://innersourcecommons.org/learn/books/getting-started-with-innersource/)<br />
```

**著者:** Andy Oram<br />
**発行日:** 2015年7月<br />
**発行元:** O’Reilly Media<br />

> **翻訳:** InnerSource Commos Japan

Reviewer: 脱字よ！

```suggestion
**翻訳:** InnerSource Commons Japan
```

## まえがき

> インナーソースの世界へようこそ。インナーソースは、オープンソースソフトウェアを単一の組織内で開発する強力なムーブメントで、プロジェクトの "オープン性" は企業内のチームを超えて広がっています。この本では、オライリーの編集者兼著者である Andy Oram が、オープンソース開発を成功させる原則を確認するとともに、大手ECのPayPalでインナーソースがどのように機能したのかを説明することによって、皆様をインナーソースの世界に導きます。

Reviewer: > オープンソースソフトウェアを単一の組織内で開発する強力なムーブメントで

文面からすると、世界に一般公開する OSS を開発することに見えます。
しかし、本編からは、企業内で共有するソフトウェアを開発し、それが OSS にもなりうることだと読み取れます。
within the walls も明示的に訳した方がそれが伝わると思います。

また、オープン性は **広がっている** というよりも、インナーソースが **広げている** という方が正確です。広げられる powerful movement なので、この本で皆にすすめたいというわけですね。

と考え、以下の訳を提案します。

```suggestion
インナーソースの世界へようこそ。インナーソースは、閉じた組織におけるオープンソースソフトウェアを開発するための原動力であり、プロジェクトの "オープン性" を企業内のチームを超えて広げるものです。この本では、オライリーの編集者兼著者である Andy Oram が、オープンソース開発を成功させる原則を確認するとともに、大手 EC の PayPal でインナーソースがどのように機能したのかを説明することによって、皆様をインナーソースの世界に導きます。
```
